SWATCH-2195: Do not wait for hourly tally operation to finish when synchronous_request is true

Sgitario opened the pull request:
Jira issue: [SWATCH-2195](https://issues.redhat.com/browse/SWATCH-2195)

## Description
Currently when we are trying to run hourly tally operation on large orgs with synchronous_request: true, we are seeing 504 Server Error: Gateway Time-out for url. We just need a way to skip queues and not to wait for response till operation finish to avoid 504 gateway timeouts.

The idea is to reuse the executor task queue that was in place along with the kafka task queue.  Before these changes, only one task queue was started (when using the kafka-queue profile, it was reading tasks from the topic). After these changes, the in memory task queue will always be started and will be used when running the hourly tally operation when the header "x-rh-swatch-use-thread-pool-executor:true" is provided.

## Testing
1. podman-compose up
2. start the service by enabling synchronous requests:
```
DEV_MODE=true ./gradlew :bootRun
```
3. Run hourly tally with the header "x-rh-swatch-use-thread-pool-executor:true":
```
http POST ":8000/api/rhsm-subscriptions/v1/internal/tally/hourly?org=16805042&start=2023-12-15T06%3A00%3A00Z&end=2023-12-15T10%3A00%3A00Z" x-rh-swatch-psk:placeholder Origin:console.redhat.com x-rh-swatch-use-thread-pool-executor:true
```
The request should have been processed and the kafka queue should be empty: http://localhost:3030/#/cluster/default/topic/n/platform.rhsm-subscriptions.tasks/

4. Run again the hourly tally with the header "x-rh-swatch-use-thread-pool-executor:false":
```
http POST ":8000/api/rhsm-subscriptions/v1/internal/tally/hourly?org=16805042&start=2023-12-15T06%3A00%3A00Z&end=2023-12-15T10%3A00%3A00Z" x-rh-swatch-psk:placeholder Origin:console.redhat.com x-rh-swatch-use-thread-pool-executor:false
```

Now, the request should go through Kafka and you should see a message in the topic: http://localhost:3030/#/cluster/default/topic/n/platform.rhsm-subscriptions.tasks/

kahowell reviewed (changes requested):
I like the solution. However, given the request is processed asynchronously, I think we should rename the flag and the header appropriately. Maybe `ENABLE_TEST_QUEUE` and `x-rh-swatch-test-queue`?

Sgitario commented:
> I like the solution. However, given the request is processed asynchronously, I think we should rename the flag and the header appropriately. Maybe `ENABLE_TEST_QUEUE` and `x-rh-swatch-test-queue`?

I had thought about it, but didn't want QE to use another header, but it makes sense to me. 
Added `x-rh-swatch-use-thread-pool-executor` and updated the PR description.

ntkathole commented:
/retest

ntkathole reviewed (approved):
Looks good 👍 

I can test 504 gateway timeout issue on stage after merge, but seems functionality wise it's working fine. 

With `x-rh-swatch-use-thread-pool-executor: false` : 
Log shows 
```
2024-03-06 11:20:20,865 [thread=http-nio-8000-exec-3] [INFO ] [org.candlepin.subscriptions.tally.job.CaptureSnapshotsTaskManager] self- Queuing hourly snapshot production for orgId 16805042 between 2023-12-15T06:00:00Z and 2023-12-15T10:00:00Z
2024-03-06 11:20:20,865 [thread=http-nio-8000-exec-3] [INFO ] [org.candlepin.subscriptions.task.queue.kafka.KafkaTaskQueue] self- Queuing task: TaskDescriptor[groupId: platform.rhsm-subscriptions.tasks, taskType: UPDATE_HOURLY_SNAPSHOTS, args: [startDateTime: [2023-12-15T06:00:00Z], endDateTime: [2023-12-15T10:00:00Z], orgId: [16805042]]]
2024-03-06 11:20:20,872 [thread=rhsm-subscriptions-task-processor-0-C-1] [INFO ] [org.candlepin.subscriptions.task.queue.kafka.KafkaTaskProcessor] - Message received from kafka (key=16805042): JsonTaskMessage(groupId=platform.rhsm-subscriptions.tasks, type=UPDATE_HOURLY_SNAPSHOTS, args={startDateTime=[2023-12-15T06:00:00Z], endDateTime=[2023-12-15T10:00:00Z], orgId=[16805042]})
```

With `x-rh-swatch-use-thread-pool-executor: true` :

The Kafka queue is skipped.
